examples/src/test/java/io/grpc/examples/header/HeaderServerInterceptorTest.java

```diff
@@ -78,17 +75,21 @@ public void sayHello(HelloRequest request, StreamObserver<HelloReply> responseOb
   @Test
   public void serverHeaderDeliveredToClient() {
     class SpyingClientInterceptor implements ClientInterceptor {
-      ClientCall.Listener<?> spyListener;
+      Metadata capturedMetadata;
 
       @Override
       public <ReqT, RespT> ClientCall<ReqT, RespT> interceptCall(
           MethodDescriptor<ReqT, RespT> method, CallOptions callOptions, Channel next) {
         return new SimpleForwardingClientCall<ReqT, RespT>(next.newCall(method, callOptions)) {
           @Override
           public void start(Listener<RespT> responseListener, Metadata headers) {
-            spyListener = responseListener =
```

Comment:
Restructuring of `ForwardingClientCallListener` and `PartialForwardingClientCallListener` broke the spy, probably because the method we want is now an inherited method. 

This works but is less obvious than the non spy():
```
spy(new SimpleForwardingClientCallListener(responseListener) {
  @Override
  public void onHeaders(Metadata headers) {
    super.onHeaders(headers);
  }
});
```

Reply:
It is broken by PartialForwardingClientCallListener being package-private. Making it public fixes it. But it's also fixed if I bump mockito to 2.13.0.

If we use AdditionalAnswers.delegatesTo instead of spy (which is what we should do anywhere we use spy, really), then it starts working:
```java
            spyListener = responseListener =
                mock(ClientCall.Listener.class,
                    delegatesTo(new SimpleForwardingClientCallListener(responseListener) {}));
```

Although then it can be simplified to:
```java
            spyListener = responseListener =
                mock(ClientCall.Listener.class, delegatesTo(responseListener));
```